### PR TITLE
Document releases and tags

### LINKED ISSUE #27: Handling different Alertmanager API variants
Alertmanager API isn't stable yet, which means that `0.4` branch API is incompatible with `0.5`.
Let's cut `0.1` unsee branch that targets Alertmanager `0.4` and `0.2` branch that targets `0.5` API.
With that we can release `v0.1.0` of the `0.1` branch and start work on `v0.2.0`.
After that let's update `README` and document which unsee one needs for given Alertmanager version.

### PR DESCRIPTION
**Pull request by prymitive**
Fixes #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,16 +29,26 @@ unsee is go installable, so the easiest way is to run:
...
     git clone https://github.com/cloudflare/unsee $GOPATH/src/github.com/cloudflare/unsee
     cd $GOPATH/src/github.com/cloudflare/unsee
-    make
 
-`unsee` binary will be compiled in project directory.
+Check [supported Alertmanager versions](#supported-alertmanager-versions) and if
+needed switch to the release branch that supports your Alertmanager version.
+For example to build unsee `0.2.x` run:
+
+    git checkout -t origin/release-0.2
```
Comment (jamesog) [inline README.md]: Do you need `origin/` here? I thought when you set it to track it will do the right thing.

File: README.md [hunk 2]
```diff
@@ -18,16 +29,26 @@ unsee is go installable, so the easiest way is to run:
     go install github.com/cloudflare/unsee
 
 The `unsee` binary will be installed into `$GOPATH/bin` directory.
+Note that this will build latest master version which might not work with your
```
Comment (jamesog) [inline README.md]: "the latest master version"

File: README.md [hunk 3]
```diff
@@ -9,6 +9,17 @@ to alert data, therefore safe to be accessed by wider audience.
 
 ![Screenshot](/screenshot.png)
 
+## Supported Alertmanager versions
+
+Alertmanager API isn't stable yet and can change between releases, make sure
```
Comment (jamesog) [inline README.md]: Either "The Alertmanager API" or "Alertmanager's API".

File: README.md [hunk 4]
```diff
@@ -9,6 +9,17 @@ to alert data, therefore safe to be accessed by wider audience.
...
+that correct version of unsee is used:
```
Comment (jamesog) [inline README.md]: Either "the correct version" or "that the correct version".

**Comment by Tenzer** [comment]
I think, realistically, we'll only really maintain a version of unsee which uses the version of Alertmanager we are using. That version should probably be in the `master` branch in order to make the workflow easier.

Before we make changes for a newer version of Alertmanager which would break backwards incompatibility, we should make sure to create a branch ie. named `alertmanager-0.4` or based on the release version which tracks that version of Alertmanager, as that allows us to backport changes to the branch if need be later on.

I don't think there's much gain from having both `master` and `alertmanager-0.5`/`release-0.2` to keep in sync, it just seems like extra work.

Also, I'm more a fan of making the branches named after the Alertmanager they work with instead of tying it to version numbers of unsee, as we might make a breaking change/rewrite of the project at some later point where we want to bump the major version, but that then would signify that it supports a newer version of Alertmanager. That would not mean we can't bump the major version when we switch support to a newer Alertmanager version though.

File: README.md (hunk 1 above)
Comment (prymitive) [inline README.md]: I think that depends on the git version used, latest might be ok without `origin`

**Comment by prymitive** [comment]
I'm not sure if I follow @Tenzer 
* `master` is latest dev, it's always the latest code, whatever that means in term of Alertmanager compatibility
* release branches are used to make releases, they are the commits that should go into a release, you create `release-0.1` branch when you want to tag `v0.1.0` release (branch then tag latest on that branch), `release-0.1` should get all commits that should end up on `v0.1.1` so we can cut that tag and make a release.

Naming release branches after Alertmanager version, when we don't tie our versioning to Alertmanager, would just be confusing. On top of that you can have multiple Alertmanager releases that have same API and unsee can work with all of them, which `alertmanager-X` branch would you use?
Let's not over complicate this.

**Comment by prymitive** [comment]
Any concerns to address here? I wanted to cut `release-0.1` branch and tag `v0.1.0` on it.
Once this is done we can merge #23

**Pull request merged**